scripts/pre_commit_linter.py

```diff
@@ -630,1297 +872,1106 @@ def _lint_py_files(config_pylint, config_pycodestyle, files_to_lint, result):
-        and filename.endswith('.js')]
-    failed = False
-    summary_messages = []
-    # For RegExp explanation, please see https://regex101.com/r/gU7oT6/37.
-    pattern_to_match = (
-        r'templateUrl: UrlInterpolationService\.[A-z\(]+' +
-        r'(?P<directive_name>[^\)]+)')
-    with _redirect_stdout(_TARGET_STDOUT):
-        for filename in files_to_check:
-            content = FileCache.read(filename)
-            total_files_checked += 1
-            matched_patterns = re.findall(pattern_to_match, content)
-            for matched_pattern in matched_patterns:
-                matched_pattern = matched_pattern.split()
-                directive_filename = ''.join(matched_pattern).replace(
-                    '\'', '').replace('+', '')
-                if not directive_filename.endswith('_directive.html'):
-                    failed = True
-                    total_error_count += 1
-                    print (
-                        '%s --> Please ensure that this file ends'
-                        'with _directive.html.' % directive_filename)
-                    print ''
+    def check_comments(self):
+        """This function ensures that comments follow correct style."""
+        print 'Starting comment checks'
+        print '----------------------------------------'
+        summary_messages = []
+        files_to_check = [
+            filename for filename in self.all_files if not
+            any(fnmatch.fnmatch(filename, pattern) for pattern in
+                EXCLUDED_PATHS) and filename.endswith('.py')]
+        message = 'There should be a period at the end of the comment.'
+        failed = False
+        space_regex = re.compile(r'^#[^\s].*$')
+        capital_regex = re.compile('^# [a-z][A-Za-z]* .*$')
+        with _redirect_stdout(_TARGET_STDOUT):
+            for filename in files_to_check:
+                file_content = FileCache.readlines(filename)
+                file_length = len(file_content)
+                for line_num in range(file_length):
+                    line = file_content[line_num].strip()
+                    next_line = ''
+                    previous_line = ''
+                    if line_num + 1 < file_length:
+                        next_line = file_content[line_num + 1].strip()
+                    if line_num > 0:
+                        previous_line = file_content[line_num - 1].strip()
+
+                    if line.startswith('#') and not next_line.startswith('#'):
+                        # Check that the comment ends with the proper
+                        # punctuation.
+                        last_char_is_invalid = line[-1] not in (
+                            ALLOWED_TERMINATING_PUNCTUATIONS)
+                        no_word_is_present_in_excluded_phrases = not any(
+                            word in line for word in EXCLUDED_PHRASES)
+                        if last_char_is_invalid and (
+                                no_word_is_present_in_excluded_phrases)
```

Comment:
All methods and function names should start with verbs and the names should describe clearly what the method/function does. "pre_commit_linter" does not do so and should be modified accordingly.

Reply:
Done, changed the name and also docstring.